packages/lwc/index.d.ts

```diff
@@ -179,6 +187,30 @@ declare module 'lwc' {
          * Called when a descendant component throws an error in one of its lifecycle hooks
          */
         errorCallback(error: Error, stack: string): void;
+        /**
+         * Called when a Form-Associated Custom Element (FACE) is associated with an HTMLFormElement.
+         * @param form HTMLFormElement - the associated form element
+         * @see https://web.dev/articles/more-capable-form-controls#void_formassociatedcallbackform
+         */
+        formAssociatedCallback(form: HTMLFormElement | null): void;
+        /**
+         * Called when a Form-Associated Custom Element (FACE) has a disabled state that has changed.
+         * @param disabled boolean - the new disabled state
+         * @see https://web.dev/articles/more-capable-form-controls#void_formdisabledcallbackdisabled
+         */
+        formDisabledCallback(disabled: boolean): void;
+        /**
+         * Called when a Form-Associated Custom Element (FACE) has an associated form that is reset.
+         * @see https://web.dev/articles/more-capable-form-controls#void_formresetcallback
+         */
+        formResetCallback(): void;
+        /**
+         * Called when a Form-Associated Custom Element (FACE) has an associated form that is restored.
+         * @param state FormRestoreState - the state of the form during restoration
+         * @param reason FormRestoreReason - the reason the form was restored
+         * @see https://web.dev/articles/more-capable-form-controls#void_formstaterestorecallbackstate_mode
+         */
+        formStateRestoreCallback(state: FormRestoreState | null, reason: FormRestoreReason): void;
```

Comment:
The article names the second argument as `mode` but I think based on the [spec](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-reactions) `reason` is a better/more intuitive name!

Reply:
Yeah I got this from https://github.com/material-components/material-web/blob/ffc08d1/labs/behaviors/form-associated.ts. `reason` sounds like a more accurate name.

BTW after reading [web.dev a bit more](https://web.dev/articles/more-capable-form-controls#restoring-form-state) and [the material tests](https://github.com/material-components/material-web/blob/ffc08d19088ca7b244c55aefbfb3aae92eac8e0c/testing/forms.ts#L205-L206), I think we may actually be able to test `formStateRestoreCallback`. I can do that in a separate PR.

Reply:
Ah wait nevermind, they are [just simulating the event](https://github.com/material-components/material-web/blob/ffc08d19088ca7b244c55aefbfb3aae92eac8e0c/testing/forms.ts#L232) 😅.

Reply:
> Ah wait nevermind, they are [just simulating the event](https://github.com/material-components/material-web/blob/ffc08d19088ca7b244c55aefbfb3aae92eac8e0c/testing/forms.ts#L232) 😅.

Hrrm do you think it'd be worth simulating in our tests too?

Reply:
I would rather do it in a WebDriver test or something so that we can actually trigger the real browser behavior. Maybe it's possible there.